AppController/test/tc_djinn.rb

```diff
@@ -469,368 +467,6 @@ def test_start_roles_on_nodes_bad_input
-      with(/\Assh.* root@1.2.3.4 'rm -rf \/tmp\/monit/).and_return('')
-    flexmock(Kernel).should_receive(:system).
-      with(/\Assh.* root@1.2.3.5 'rm -rf \/tmp\/monit/).and_return('')
-    flexmock(Djinn).should_receive(:log_run).with(/\Arm -rf \/tmp\/monit/).
-      and_return()
-
-    # and assume that the appcontrollers start up fine
-    flexmock(HelperFunctions).should_receive(:is_port_open?).
-      with('1.2.3.4', Djinn::SERVER_PORT, HelperFunctions::USE_SSL).
-      and_return(true)
-    flexmock(HelperFunctions).should_receive(:is_port_open?).
-      with('1.2.3.5', Djinn::SERVER_PORT, HelperFunctions::USE_SSL).
-      and_return(true)
-
-    original_node_info = {
-      "public_ip" => "1.2.3.3",
-      "private_ip" => "1.2.3.3",
-      "jobs" => ["shadow"],
-      "instance_id" => "i-000000"
-    }
-
-    node1_info = {
-      "public_ip" => "1.2.3.4",
-      "private_ip" => "1.2.3.4",
-      "jobs" => ["appengine"],
-      "instance_id" => "i-ABCDEFG",
-      "disk" => nil
-    }
-
-    node2_info = {
-      "public_ip" => "1.2.3.5",
-      "private_ip" => "1.2.3.5",
-      "jobs" => ["appengine"],
-      "instance_id" => "i-HIJKLMN",
-      "disk" => nil
-    }
-
-    original_node = DjinnJobData.new(original_node_info, "boo")
-    new_node1 = DjinnJobData.new(node1_info, "boo")
-    new_node2 = DjinnJobData.new(node2_info, "boo")
-    all_nodes_serialized = JSON.dump([original_node.to_hash(),
-      new_node1.to_hash(), new_node2.to_hash()])
-
-    options = {'keyname' => 'boo', 'user_commands' => []}
-    options_as_array = options.to_a.flatten
-
-    # and that the appcontrollers receive the initial message to start
-    # up from our appcontroller
-    flexmock(AppControllerClient).new_instances { |instance|
-      instance.should_receive(:set_parameters).with(all_nodes_serialized,
-        options_as_array).and_return("OK")
-    }
-
-    # lastly, the appcontroller will update its local /etc/hosts file
-    # and /etc/hostname file with info about the new node and its own
-    # node
-    flexmock(File).should_receive(:open).with("/etc/hosts", "w+", Proc).
-      and_return()
-    flexmock(File).should_receive(:open).with("/etc/hostname", "w+", Proc).
-      and_return()
-    flexmock(Djinn).should_receive(:log_run).with("/bin/hostname appscale-image0").
-      and_return()
-
-    # mock out updating the firewall config
-    ip_list = "#{Djinn::APPSCALE_CONFIG_DIR}/all_ips"
-    flexmock(File).should_receive(:open).with(ip_list, "w+", Proc).and_return()
-    flexmock(Djinn).should_receive(:log_run).with(/bash .*firewall.conf/)
-
-    djinn = Djinn.new()
-    djinn.nodes = [original_node]
-    djinn.my_index = 0
-    djinn.options = options
-    flexmock(djinn).should_receive(:add_nodes).and_return()
-    actual = djinn.start_new_roles_on_nodes_in_cloud(ips_to_roles)
-    assert_equal(true, actual.include?(node1_info))
-    assert_equal(true, actual.include?(node2_info))
-  end
```

Comment:
If these unit tests are no longer useful, should they be replaced with one that is?